### ISSUE TITLE
Possibly unbounded variable not found

### ISSUE THREAD

**Issue by alonme** [pylint-dev/pylint#8783]
### Bug description

```
text = input()

if text == "testing":
    unbounded = True

print(unbounded)
```

For some reason `unbounded` isn't caught by `E0601` (or other rule.. but it should be caught by something...)

If a simpler condition is used the problem doesn't happen

```
if 1 == 2:
    unbounded = True

print(unbounded)
```

### Command used

```shell
pylint test.py --disable=missing-module-docstring,invalid-name
```

### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

I would expect pylint to detect E0601 on this code

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.8.7 (default, Oct 29 2022, 08:49:12)
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Will be marking as duplicate of #1727. `pylint` has more reusable utilities to facilitate implementing this now, in 2023, than when it was first proposed 👍
